gdnative-core/src/variant.rs

```diff
@@ -813,15 +959,47 @@ impl ToVariant for () {
 }
 
 impl FromVariant for () {
-    fn from_variant(variant: &Variant) -> Option<Self> {
-        if variant.get_type() == VariantType::Nil {
-            Some(())
-        } else {
-            None
-        }
+    fn from_variant(variant: &Variant) -> Result<Self, FromVariantError> {
+        variant.try_as_sys_of_type(VariantType::Nil).map(|_| ())
     }
 }
 
+macro_rules! from_variant_transmute {
+    (
+        $(
+            impl FromVariant for $TryType:ident : $try_gd_method:ident;
+        )*
+    ) => (
+        $(
+            impl FromVariant for $TryType {
+                fn from_variant(variant: &Variant) -> Result<Self, FromVariantError> {
+                    unsafe {
+                        variant.try_as_sys_of_type(VariantType::$TryType)
+                            .map(|v| (get_api().$try_gd_method)(v))
+                            .map(|v| transmute(v))
```

Comment:
Yes for some types. Implementation updated for them.

We still have to use `transmute` for `Vector2`, `Vector3`, `Quat`, `Rect2` and `Transform2D` since they're type aliases.

Reply:
Right, thanks!